BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -510,39 +534,48 @@ extension ConvivaAnalytics: BitmovinPlayerListenerDelegate {
 
     // MARK: - Ad events
     func onAdStarted(_ event: AdStartedEvent) {
+        var adInfo = event.ad.adInfo
         let adPosition: ConvivaSDK.AdPosition = AdEventUtil.parseAdPosition(
             event: event,
             contentDuration: player.duration
         )
-        var adAttributes = [String: Any]()
-        adAttributes["c3.ad.position"] = adPosition.rawValue
-        videoAnalytics.reportAdBreakStarted(
-            AdPlayer.ADPLAYER_CONTENT,
-            adType: AdTechnology.CLIENT_SIDE,
-            adBreakInfo: adAttributes
+        adInfo["c3.ad.position"] = adPosition.rawValue
+
+        adAnalytics.reportAdLoaded(adInfo)
+        adAnalytics.reportAdStarted(adInfo)
+        adAnalytics.reportAdMetric(
+            CIS_SSDK_PLAYBACK_METRIC_PLAYER_STATE,
+            value: PlayerState.CONVIVA_PLAYING.rawValue
         )
     }
 
     func onAdFinished() {
-        videoAnalytics.reportAdBreakEnded()
+        adAnalytics.reportAdEnded()
     }
 
     func onAdSkipped(_ event: AdSkippedEvent) {
-        customEvent(event: event)
-        videoAnalytics.reportAdBreakEnded()
+        adAnalytics.reportAdSkipped()
     }
 
     func onAdError(_ event: AdErrorEvent) {
-        customEvent(event: event)
-        videoAnalytics.reportAdBreakEnded()
+        adAnalytics.reportAdFailed(event.message, adInfo: nil)
     }
 
     func onAdBreakStarted(_ event: AdBreakStartedEvent) {
-        customEvent(event: event)
+        var adAttributes = [String: Any]()
+        adAttributes["c3.ad.position"] = AdEventUtil.parseAdPosition(
+            event: event,
+            contentDuration: player.duration
+        ).rawValue
+        videoAnalytics.reportAdBreakStarted(
+            AdPlayer.ADPLAYER_CONTENT,
+            adType: AdTechnology.CLIENT_SIDE,
+            adBreakInfo: adAttributes
+        )
     }
 
     func onAdBreakFinished(_ event: AdBreakFinishedEvent) {
-        customEvent(event: event)
+        videoAnalytics.reportAdBreakEnded()
     }
 
     func onDestroy() {
```

Comment:
I can't comment on `internalEndSession()`:

Should we also end the `adAnalytics` in there? See [this](https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/104#discussion_r1625596100) change on the web integration.

Reply:
I am not sure. I couldn't find anything in that regard in the documentation.
I also looked at another integration, and this is not done there.